lib/srv/db/common/srvconnect/srvconnect.go

```diff
@@ -0,0 +1,299 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package srvconnect
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"log/slog"
+	"math/rand"
+	"net"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	apidefaults "github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/cryptosuites"
+	"github.com/gravitational/teleport/lib/reversetunnelclient"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/tlsca"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// DatabaseServersGetter is an interface for retrieving information about
+// database proxy servers within a specific namespace.
+type DatabaseServersGetter interface {
+	// GetDatabaseServers returns all registered database proxy servers.
+	GetDatabaseServers(ctx context.Context, namespace string, opts ...services.MarshalOption) ([]types.DatabaseServer, error)
+}
+
+// Cluster represents a remote site the behavior of a cluster in the system,
+// providing methods to retrieve its name and access point.
+type Cluster interface {
+	// GetName returns site name (identified by authority domain's name).
+	GetName() string
+}
+
+// GetDatabaseServersParams contains the parameters required to retrieve
+// database servers from a specific cluster.
+type GetDatabaseServersParams struct {
+	Logger *slog.Logger
+	// Cluster represents the cluster to which the database belongs.
+	Cluster Cluster
```

Comment:
Why use an interface instead of the string? Couldn't we get the cluster from Identity.RouteToCluster?

Reply:
just curious if this cluster name will be different from `Identity.RouteToCluster`?